src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentLoweringPass.cs

```diff
@@ -234,7 +241,7 @@ private ComponentChildContentIntermediateNode RewriteChildContent(BoundAttribute
                     }
                     else if (child is TagHelperPropertyIntermediateNode property)
                     {
-                        if (property.BoundAttribute.Kind == BlazorMetadata.ChildContent.TagHelperKind)
+                        if (property.BoundAttribute.IsChildContentParameterNameProperty())
```

Comment:
This check is stricter than it was before, which is nice. If we end up adding anything additional to a 'child content' element, then this code won't have to change too much.

Reply:
Previously we assumed that any 'component' parameter on a child content element was 'Context'.  Now we're matching on metadata.